src/ui/react/hooks.ts

```diff
@@ -45,7 +47,11 @@ function makeUpdater<Domain, K extends keyof Domain>(
   const updater: UpdaterInternal<Domain[K]> = function(options, delta) {
     massUpdater(options, kind, id, delta);
   };
-  return (a: any, b?: any) => updater(b ? a : {}, b || a);
+  return ((a: any, b?: any) => {
+    if (typeof a === "string")
+      return updater({}, prev => ({ ...prev, [a]: b }));
+    updater(b ? a : {}, b || a);
```

Comment:
In this scenario (where `typeof a !== "string"`), then what is `b`? I thought prior to this patch form it only took one argument, the delta function.

Reply:
From the very beginning the delta function form supports the options parameter before it: `updateThing({ ... }, deltaF)`. I just don’t have any options yet, so it’s not documented, but a lot of my future designs imply needing the options.